### ISSUE TITLE
Interface proposal, variation 1

### ISSUE THREAD

**Issue by pmjones** [payload-interop/payload-interop#7]
Based on researched commonalities among existing implementations, I propose the following method interface, and status constants class:

```php
namespace PayloadInterop;

interface DomainPayload
{
    /**
     * Returns the domain status.
     */
    public function getStatus() : string;

    /**
     * Returns the input as recognized by the domain.
     */
    public function getInput() : array;

    /**
     * Returns the output produced by the domain,
     * generally entities or aggregates.
     */
    public function getOutput() : array;

    /**
     * Returns any messages produced by the domain,
     * generally error or invalidation messages.
     */
    public function getMessages() : array;
}

class DomainStatus
{
    /**
     * Input accepted for later processing.
     */
    const ACCEPTED = 'ACCEPTED';

    /**
     * Create succeeded.
     */
    const CREATED = 'CREATED';

    /**
     * Delete succeeded.
     */
    const DELETED = 'DELETED';

    /**
     * Generic error.
     */
    const ERROR = 'ERROR';

    /**
     * Find succeeded.
     */
    const FOUND = 'FOUND';

    /**
     * Input not valid.
     */
    const INVALID = 'INVALID';

    /**
     * Find failed (n.b.: not an error).
     */
    const NOT_FOUND = 'NOT_FOUND';

    /**
     * Processing is not finished yet.
     */
    const PROCESSING = 'PROCESSING';

    /**
     * Generic success.
     */
    const SUCCESS = 'SUCCESS';

    /**
     * Interaction is not authorized.
     */
    const UNAUTHORIZED = 'UNAUTHORIZED';

    /**
     * Update succeeded.
     */
    const UPDATED = 'UPDATED';

    /**
     * Disallow instantiation.
     */
    final private function __construct() {}
}
```

**Comment by niden** [comment]
We discussed in a previous thread the `getException()` which would hold the particular exception (if applicable)

Do you envision this information to be included in the `getMessages()` according to the needs of the developer?

**Comment by pmjones** [comment]
Hi @niden -- in the end, I think adding a method that doesn't actually exist in many of the reference implementations is unwarranted. I'm not *against* it, exactly, it just doesn't appear that anyone else needed it very much.

> Do you envision this information to be included in the getMessages() according to the needs of the developer?

Either `getMessages()`, or `getOutput()`.

**Comment by pmjones** [comment]
Closed in favor of #8.
